test/optimizer/plugins/ResolveContextURLsOptimizer.test.ts

```diff
@@ -0,0 +1,195 @@
+      const extension = new ExportableExtension('MyExtension');
+      extension.contexts = [
+        {
+          isQuoted: false,
+          value: 'https://demo.org/StructureDefinition/SmallPatient#name'
+        }
+      ];
+      const myPackage = new Package();
+      myPackage.add(extension);
+      optimizer.optimize(myPackage, fisher);
+      expect(extension.contexts).toEqual([
+        {
+          isQuoted: false,
+          value: 'SmallPatient.name'
+        }
+      ]);
+    });
+
+    it('should replace an unquoted context with a resource name and no path when the context value is a URL of a known resource no path', () => {
+      const extension = new ExportableExtension('MyExtension');
+      extension.contexts = [
+        {
+          isQuoted: false,
+          value: 'https://demo.org/StructureDefinition/SmallPatient'
+        }
+      ];
+      const myPackage = new Package();
+      myPackage.add(extension);
+      optimizer.optimize(myPackage, fisher);
+      expect(extension.contexts).toEqual([
+        {
+          isQuoted: false,
+          value: 'SmallPatient'
+        }
+      ]);
+    });
+
+    it('should replace an unquoted context with an alias when the context value is a URL of an unknown resource', () => {
+      const extension = new ExportableExtension('MyExtension');
+      extension.contexts = [
+        {
+          isQuoted: false,
+          value: 'https://demo.org/StructureDefinition/MysteriousExtension'
+        }
+      ];
+      const myPackage = new Package();
+      myPackage.add(extension);
+      optimizer.optimize(myPackage, fisher);
+      expect(extension.contexts).toEqual([
+        {
+          isQuoted: false,
+          value: '$MysteriousExtension'
+        }
+      ]);
+      expect(myPackage.aliases).toEqual([
+        {
+          alias: '$MysteriousExtension',
+          url: 'https://demo.org/StructureDefinition/MysteriousExtension'
+        }
+      ]);
+    });
+
+    it('should replace an unquoted context with an alias and path when the context value is a URL of an unknown resource with a path', () => {
+      const extension = new ExportableExtension('MyExtension');
+      extension.contexts = [
+        {
+          isQuoted: false,
+          value: 'https://demo.org/StructureDefinition/UnknownObservation#method'
+        }
+      ];
+      const myPackage = new Package();
+      myPackage.add(extension);
+      optimizer.optimize(myPackage, fisher);
+      expect(extension.contexts).toEqual([
+        {
+          isQuoted: false,
+          value: '$UnknownObservation#method'
+        }
+      ]);
+      expect(myPackage.aliases).toEqual([
+        {
+          alias: '$UnknownObservation',
+          url: 'https://demo.org/StructureDefinition/UnknownObservation'
+        }
+      ]);
+    });
+
+    it('should not change an unquoted context when the context value is a URL of an unknown resource with a path and the alias option is false', () => {
```

Comment:
Super minor, but I think the description of this test (unknown resource with a path) actually goes with the test below it and that test's description (unknown resource (no path)) goes here.